sklearn/feature_selection/_base.py

```diff
@@ -196,23 +197,44 @@ def _get_feature_importances(estimator, getter, transform_func=None, norm_order=
     """
     if isinstance(getter, str):
         if getter == "auto":
-            if hasattr(estimator, "coef_"):
-                getter = attrgetter("coef_")
-            elif hasattr(estimator, "feature_importances_"):
-                getter = attrgetter("feature_importances_")
-            else:
-                raise ValueError(
-                    "when `importance_getter=='auto'`, the underlying "
-                    f"estimator {estimator.__class__.__name__} should have "
-                    "`coef_` or `feature_importances_` attribute. Either "
-                    "pass a fitted estimator to feature selector or call fit "
-                    "before calling transform."
+            with warnings.catch_warnings():
```

Comment:
This is a big change to work around the `_PLS` warning.

Since this is a workaround is specifically for `_PLS`, I think we can do `attrgetter("_coef_")` directly and avoid catching the warning all together:

```python
if isinstance(estimator, _PLS):
    getter = attrgetter("_coef_")
elif hasattr(estimator, "coef_"):
    ...